resource/resource_test.go

```diff
@@ -636,7 +636,7 @@ func TestCheckRepoDirSuccess(t *testing.T) {
 		defer teardown(t)
 
 		t.Run(tc.name, func(t *testing.T) {
-			err := checkRepoDir(filepath.Join(inDir, tc.dir), wantOwner, wantRepo)
+			err := checkRepoDir(filepath.Join(inDir, tc.dir), wantOwner, wantRepo, "github.com")
```

Comment:
Not sure this is the best way to do this. It is strange that all the other params to `checkRepoDir` are variables, but "github.com" is a literal string...

Reply:
Note 1:
We need to update also the 3 helper functions - sshRemote(), httpsRemote() and httpRemote().
Or maybe we get rid of them and inline the sprintfs.

Note 2:
> Not sure this is the best way to do this. It is strange that all the other params to checkRepoDir are variables, but "github.com" is a literal string...

Indeed. We need to add a field `githubHost` to the struct, after line 610 and initialize it in each test.
We also need to add a few tests with a custom githubHost, to test the feature added in this PR.